src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py

```diff
@@ -0,0 +1,42 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Module containing the Esdl to HeatPump asset mapper class."""
+
+from omotes_simulator_core.entities.assets.heat_pump import HeatPump
+from omotes_simulator_core.entities.assets.asset_abstract import AssetAbstract
+from omotes_simulator_core.entities.assets.esdl_asset_object import EsdlAssetObject
+from omotes_simulator_core.simulation.mappers.mappers import EsdlMapperAbstract
+
+
+class EsdlAssetHeatPumpMapper(EsdlMapperAbstract):
+    """Class to map an ESDL asset to a heatpump entity class."""
+
+    def to_esdl(self, entity: HeatPump) -> EsdlAssetObject:
+        """Map a HeatPump entity to an EsdlAsset."""
+        raise NotImplementedError("EsdlAssetHeatPumpMapper.to_esdl()")
+
+    def to_entity(self, esdl_asset: EsdlAssetObject) -> AssetAbstract:
```

Comment:
This mapper will require a get COP from the model, but I will add it in the branch of the HP.

Reply:
what are COP and HP?

Reply:
Coefficient of Performance (COP) of the Heat Pump (HP) these must be added in the HP branch.